tests/value/test_semivalues.py

```diff
@@ -58,6 +60,48 @@ def test_shapley(
     check_values(values, exact_values, rtol=0.2)
 
 
+@pytest.mark.parametrize(
+    "num_samples,sampler,coefficient,batch_size",
+    [(5, PermutationSampler, beta_coefficient(1, 1), 5)],
+)
+def test_shapley_batch_size(
+    num_samples: int,
+    analytic_shapley,
+    sampler: Type[PermutationSampler],
+    coefficient: SVCoefficient,
+    batch_size: int,
+    n_jobs: int,
+    parallel_config: ParallelConfig,
+    seed: Seed,
+):
+    u, exact_values = analytic_shapley
+    criterion = AbsoluteStandardError(0.02, 1.0) | MaxUpdates(2 ** (num_samples * 2))
+    result_single_batch, total_seconds_single_batch = measure_execution_time(
+        compute_generic_semivalues
+    )(
+        sampler(u.data.indices, seed=seed),
+        u,
+        coefficient,
+        criterion,
+        n_jobs=n_jobs,
+        batch_size=1,
+        config=parallel_config,
+    )
+    result_multi_batch, total_seconds_multi_batch = measure_execution_time(
+        compute_generic_semivalues
+    )(
+        sampler(u.data.indices, seed=seed),
+        u,
+        coefficient,
+        criterion,
+        n_jobs=n_jobs,
+        batch_size=batch_size,
+        config=parallel_config,
+    )
+    assert total_seconds_multi_batch < total_seconds_single_batch
+    check_values(result_single_batch, result_multi_batch, rtol=0.0, atol=0.0)
```

Comment:
Loosed it a little bit.

Reply:
I meant that the sample which triggers the stopping criterion can be different when using batches, despite the fact that you check done() on a sample by sample basis. Consequently a check with 0 tolerance can fail.

Reply:
It can be indeed different, once batch_2 arrives before process batch_1. The order inside of a batch is preserved. Relaxed the relative tolerance to `1e-4` and added a comment in case the test case fails in the future.